spec/features/pictures_feature_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'rails_helper'
+
+feature 'Pictures' do
+
+  let(:user) { build :user }
+  before { register(user) }
+
+  context 'when there are no pictures' do
+
+    scenario 'should have a prompt to add a picture' do
+      expect(page).to have_content 'Add a post'
+    end
+
+    scenario 'cannot add a picture if not logged in' do
+      click_link 'Log out'
+      expect(page).to have_content 'Please log in or register to upload a picture.'
+    end
+
+    scenario 'can add a picture if logged in' do
+      add_picture
+      expect(page).to have_content 'Duck Punching'
+      expect(page).not_to have_content 'No pictures yet'
+    end
+
+  end
+
+  context 'when creating pictures' do
+
+    scenario "picture's name cannot be shorter than 2 characters" do
+      visit '/'
+      click_link 'New Post'
+      attach_file('picture[image]', File.join(Rails.root,'spec',"files", 'images', 'duck.jpg'))
```

Comment:
Metrics/LineLength: Line is too long. [95/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)